In [343]:
# [1] 모듈 로딩
import pandas as pd
import numpy as np

In [344]:
file_name='소비자물가상승률_2013.2023.xlsx'

#### [소비자 물가 지수와 중고차 경매 시장 관계 분석]
- 사용한 데이터 : 중고차 경매시장 데이터 (2021년 1월 ~ 2024년 4월)
- [ 데이터 출처 ]
    - 경매 낙찰대수: 전국자동차경매장협회 (KAAA)
    - 중고차 거래대수: 국토교통부 자동차 등록현황

In [345]:
file_name2='2021년1월~2024년4월_경매시장관련Data.xlsx'

In [346]:
cpi=pd.read_excel(file_name, skiprows=28, nrows=14)

c:\Users\KDP-50\anaconda3\envs\py39\lib\site-packages\openpyxl\styles\stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
  warn("Workbook contains no default style, apply openpyxl's default")


In [347]:
cpi.rename(columns={'Unnamed: 0':'목록'}, inplace=True)
cpi

목록  2013  2014  2015  2016  2017  2018  2019   2020  \
0   소비자물가 총지수(2020=100) 93.00 94.20 94.90 95.80 97.60 99.10 99.50 100.00   
1           소비자물가상승률(%)  1.30  1.30  0.70  1.00  1.90  1.50  0.40   0.50   
2           식료품 · 비주류음료  0.90  0.30  1.60  2.30  3.40  2.80  0.00   4.40   
3               주류 및 담배  1.70 -0.10 50.10  0.70  1.50  0.30  0.60   0.30   
4               의류 및 신발  2.90  4.00  1.30  1.80  1.10  1.10  0.10   0.70   
5      주택 ·수도 · 전기 및 연료  3.50  2.90 -0.60 -0.80  1.70  0.70  1.20   0.50   
6         가정용품 및 가사 서비스  0.30  2.10  2.60  1.60  1.10  2.30  2.10   0.00   
7                    보건  0.40  0.70  1.30  1.00  0.90 -0.10  0.50   1.50   
8                    교통 -0.50 -1.60 -7.80 -2.20  3.60  2.40 -1.80  -1.80   
9                    통신 -0.10 -0.10 -0.20  0.10  0.30 -0.90 -2.30  -2.10   
10              오락 및 문화  1.00  0.40 -0.50  1.80  0.10  0.50 -0.20  -1.00   
11                   교육  1.20  1.50  1.70  1.60  1.10  1.40  0.50  -2.10   
12              음식 및 숙박  1.60  1.50  2.30  2.50  2.40  3.00  1.80   0.90   
13          기타 상품 및 서비스  0.50  3.10  2.70  3.40  2.80  0.60  1.60   2.00   

     2021   2022   2023  
0  102.50 107.70 111.60  
1    2.50   5.10   3.60  
2    5.90   5.90   5.50  
3    0.40   2.20   1.00  
4    0.60   3.10   6.70  
5    1.60   5.50   5.00  
6    1.90   4.80   5.40  
7   -0.10   0.80   1.60  
8    6.30   9.70  -3.40  
9   -0.90   0.90   1.00  
10   0.40   2.80   3.70  
11   0.90   1.30   2.00  
12   2.70   7.60   6.00  
13   2.00   6.10   5.80

In [348]:
check_cpi=cpi[['목록','2021','2022','2023']]
check_cpi

목록   2021   2022   2023
0   소비자물가 총지수(2020=100) 102.50 107.70 111.60
1           소비자물가상승률(%)   2.50   5.10   3.60
2           식료품 · 비주류음료   5.90   5.90   5.50
3               주류 및 담배   0.40   2.20   1.00
4               의류 및 신발   0.60   3.10   6.70
5      주택 ·수도 · 전기 및 연료   1.60   5.50   5.00
6         가정용품 및 가사 서비스   1.90   4.80   5.40
7                    보건  -0.10   0.80   1.60
8                    교통   6.30   9.70  -3.40
9                    통신  -0.90   0.90   1.00
10              오락 및 문화   0.40   2.80   3.70
11                   교육   0.90   1.30   2.00
12              음식 및 숙박   2.70   7.60   6.00
13          기타 상품 및 서비스   2.00   6.10   5.80

In [349]:
check_cpi.set_index('목록',inplace=True)
check_cpi

2021   2022   2023
목록                                      
소비자물가 총지수(2020=100) 102.50 107.70 111.60
소비자물가상승률(%)           2.50   5.10   3.60
식료품 · 비주류음료           5.90   5.90   5.50
주류 및 담배               0.40   2.20   1.00
의류 및 신발               0.60   3.10   6.70
주택 ·수도 · 전기 및 연료      1.60   5.50   5.00
가정용품 및 가사 서비스         1.90   4.80   5.40
보건                   -0.10   0.80   1.60
교통                    6.30   9.70  -3.40
통신                   -0.90   0.90   1.00
오락 및 문화               0.40   2.80   3.70
교육                    0.90   1.30   2.00
음식 및 숙박               2.70   7.60   6.00
기타 상품 및 서비스           2.00   6.10   5.80

In [350]:
cpi.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14 entries, 0 to 13
Data columns (total 12 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   목록      14 non-null     object 
 1   2013    14 non-null     float64
 2   2014    14 non-null     float64
 3   2015    14 non-null     float64
 4   2016    14 non-null     float64
 5   2017    14 non-null     float64
 6   2018    14 non-null     float64
 7   2019    14 non-null     float64
 8   2020    14 non-null     float64
 9   2021    14 non-null     float64
 10  2022    14 non-null     float64
 11  2023    14 non-null     float64
dtypes: float64(11), object(1)
memory usage: 1.4+ KB


In [351]:
used_car_data=pd.read_excel(file_name2, nrows=28)

In [352]:
used_car_data.head(2)

구분 Unnamed: 1           내역         1월         2월         3월  \
0  신차 \n등록 대수      2021년  한국 전체 시장 합계 156,329.00 125,086.00 170,520.00   
1         NaN      2022년  한국 전체 시장 합계 133,178.00 115,339.00 145,287.00   

          4월         5월         6월         7월         8월         9월  \
0 164,449.00 146,796.00 168,155.00 160,302.00 127,129.00 121,292.00   
1 146,798.00 147,517.00 136,213.00 152,298.00 134,909.00 142,897.00   

         10월        11월        12월       합계(누계)  
0 123,745.00 143,221.00 141,955.00 1,748,979.00  
1 141,742.00 158,132.00 142,825.00 1,697,135.00

In [353]:
used_car_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 16 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   구분          3 non-null      object 
 1   Unnamed: 1  12 non-null     object 
 2   내역          28 non-null     object 
 3   1월          28 non-null     float64
 4   2월          28 non-null     float64
 5   3월          28 non-null     float64
 6   4월          28 non-null     float64
 7   5월          21 non-null     float64
 8   6월          21 non-null     float64
 9   7월          21 non-null     float64
 10  8월          21 non-null     float64
 11  9월          21 non-null     float64
 12  10월         21 non-null     float64
 13  11월         21 non-null     float64
 14  12월         21 non-null     float64
 15  합계(누계)      21 non-null     float64
dtypes: float64(13), object(3)
memory usage: 3.6+ KB


In [354]:
used_car_data.drop([0,1,2,3], inplace=True)
used_car_data.head(2)

구분 Unnamed: 1      내역        1월        2월        3월        4월  \
4  경매\n낙찰 대수      2021년  경매출품대수 25,419.00 22,800.00 24,334.00 28,117.00   
5        NaN        NaN    낙찰대수 16,659.00 15,293.00 17,719.00 19,559.00   

         5월        6월        7월        8월        9월       10월       11월  \
4 24,533.00 25,082.00 29,191.00 23,633.00 25,115.00 26,351.00 24,725.00   
5 17,032.00 17,843.00 19,860.00 16,209.00 16,676.00 17,985.00 17,532.00   

        12월     합계(누계)  
4 24,852.00 304,152.00  
5 16,634.00 209,001.00

In [355]:
used_car_data.columns

Index(['구분', 'Unnamed: 1', '내역', '1월', '2월', '3월', '4월', '5월', '6월', '7월',
       '8월', '9월', '10월', '11월', '12월', '합계(누계)'],
      dtype='object')

In [356]:
used_car_data.rename(columns={'Unnamed: 1':'연도'}, inplace=True)

In [357]:
used_car_data.head(2)

구분     연도      내역        1월        2월        3월        4월        5월  \
4  경매\n낙찰 대수  2021년  경매출품대수 25,419.00 22,800.00 24,334.00 28,117.00 24,533.00   
5        NaN    NaN    낙찰대수 16,659.00 15,293.00 17,719.00 19,559.00 17,032.00   

         6월        7월        8월        9월       10월       11월       12월  \
4 25,082.00 29,191.00 23,633.00 25,115.00 26,351.00 24,725.00 24,852.00   
5 17,843.00 19,860.00 16,209.00 16,676.00 17,985.00 17,532.00 16,634.00   

      합계(누계)  
4 304,152.00  
5 209,001.00

In [358]:
used_car_data['연도']

4     2021년
5       NaN
6       NaN
7     2022년
8       NaN
9       NaN
10    2023년
11      NaN
12      NaN
13    2024년
14      NaN
15      NaN
16    2021년
17      NaN
18      NaN
19    2022년
20      NaN
21      NaN
22    2023년
23      NaN
24      NaN
25    2024년
26      NaN
27      NaN
Name: 연도, dtype: object

In [359]:
used_car_data['구분']

4      경매\n낙찰 대수
5            NaN
6            NaN
7            NaN
8            NaN
9            NaN
10           NaN
11           NaN
12           NaN
13           NaN
14           NaN
15           NaN
16    중고차\n거래 대수
17           NaN
18           NaN
19           NaN
20           NaN
21           NaN
22           NaN
23           NaN
24           NaN
25           NaN
26           NaN
27           NaN
Name: 구분, dtype: object

In [360]:
used_car_data['구분'].iloc[0] = '경매낙찰대수' # 경매낙찰대수
used_car_data['구분'].iloc[12] = '중고차거래대수' # 중고차거래대수

C:\Users\KDP-50\AppData\Local\Temp\ipykernel_11004\812869565.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  used_car_data['구분'].iloc[0] = '경매낙찰대수' # 경매낙찰대수
C:\Users\KDP-50\AppData\Local\Temp\ipykernel_11004\812869565.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  used_car_data['구분'].iloc[12] = '중고차거래대수' # 중고차거래대수


In [361]:
used_car_data['연도']=used_car_data['연도'].ffill().copy()
used_car_data['구분']=used_car_data['구분'].ffill().copy()

In [362]:
used_car_data

구분     연도        내역         1월         2월         3월         4월  \
4    경매낙찰대수  2021년    경매출품대수  25,419.00  22,800.00  24,334.00  28,117.00   
5    경매낙찰대수  2021년      낙찰대수  16,659.00  15,293.00  17,719.00  19,559.00   
6    경매낙찰대수  2021년    낙찰율(%)       0.66       0.67       0.73       0.70   
7    경매낙찰대수  2022년    경매출품대수  26,702.00  25,953.00  33,488.00  33,256.00   
8    경매낙찰대수  2022년      낙찰대수  17,852.00  17,551.00  21,481.00  19,243.00   
9    경매낙찰대수  2022년    낙찰율(%)       0.67       0.68       0.64       0.58   
10   경매낙찰대수  2023년    경매출품대수  31,238.00  29,794.00  33,702.00  29,912.00   
11   경매낙찰대수  2023년      낙찰대수  18,603.00  20,281.00  24,501.00  20,908.00   
12   경매낙찰대수  2023년  낙찰율 (당월)       0.60       0.68       0.73       0.70   
13   경매낙찰대수  2024년    경매출품대수  33,993.00  33,089.00  33,089.00  33,818.00   
14   경매낙찰대수  2024년      낙찰대수  20,284.00  21,609.00  21,609.00  21,217.00   
15   경매낙찰대수  2024년  낙찰율 (당월)       0.60       0.65       0.65       0.63   
16  중고차거래대수  2021년    사업자 거래 194,646.00 188,033.00 247,252.00 233,060.00   
17  중고차거래대수  2021년    당사자 거래 112,122.00 105,001.00 126,106.00 117,993.00   
18  중고차거래대수  2021년       총합계 306,768.00 293,034.00 373,358.00 351,053.00   
19  중고차거래대수  2022년    사업자 거래 200,042.00 195,899.00 226,655.00 217,975.00   
20  중고차거래대수  2022년    당사자 거래 100,239.00  94,242.00 107,611.00 103,954.00   
21  중고차거래대수  2022년       총합계 300,281.00 290,141.00 334,266.00 321,929.00   
22  중고차거래대수  2023년    사업자 거래 170,191.00 195,597.00 230,898.00 203,097.00   
23  중고차거래대수  2023년    당사자 거래 109,119.00 114,579.00 132,538.00 110,158.00   
24  중고차거래대수  2023년       총합계 279,310.00 310,176.00 363,436.00 313,255.00   
25  중고차거래대수  2024년    사업자 거래 217,457.00 190,954.00 223,854.00 228,623.00   
26  중고차거래대수  2024년    당사자 거래 118,339.00 102,041.00 113,047.00 114,387.00   
27  중고차거래대수  2024년       총합계 335,796.00 292,995.00 336,901.00 343,010.00   

           5월         6월         7월         8월         9월        10월  \
4   24,533.00  25,082.00  29,191.00  23,633.00  25,115.00  26,351.00   
5   17,032.00  17,843.00  19,860.00  16,209.00  16,676.00  17,985.00   
6        0.69       0.71       0.68       0.69       0.66       0.68   
7   30,134.00  31,563.00  33,369.00  30,573.00  29,416.00  28,137.00   
8   18,493.00  19,129.00  18,728.00  18,407.00  18,819.00  17,567.00   
9        0.61       0.61       0.56       0.60       0.64       0.62   
10  29,437.00  33,268.00  34,226.00  29,801.00  26,479.00  29,500.00   
11  20,011.00  22,187.00  20,944.00  20,166.00  17,901.00  19,030.00   
12       0.68       0.67       0.61       0.68       0.68       0.65   
13        NaN        NaN        NaN        NaN        NaN        NaN   
14        NaN        NaN        NaN        NaN        NaN        NaN   
15        NaN        NaN        NaN        NaN        NaN        NaN   
16 213,026.00 227,944.00 225,245.00 210,308.00 191,549.00 208,746.00   
17 104,710.00 113,129.00 110,018.00 101,569.00  93,120.00  99,908.00   
18 317,736.00 341,073.00 335,263.00 311,877.00 284,669.00 308,654.00   
19 224,118.00 210,354.00 209,396.00 217,584.00 202,374.00 200,684.00   
20 107,049.00 107,088.00 107,014.00 107,489.00 102,398.00 100,525.00   
21 331,167.00 317,442.00 316,410.00 325,073.00 304,772.00 301,209.00   
22 211,730.00 214,056.00 206,892.00 209,065.00 191,955.00 200,088.00   
23 111,048.00 115,704.00 108,253.00 108,751.00  96,711.00 102,817.00   
24 322,778.00 329,760.00 315,145.00 317,816.00 288,666.00 302,905.00   
25        NaN        NaN        NaN        NaN        NaN        NaN   
26        NaN        NaN        NaN        NaN        NaN        NaN   
27        NaN        NaN        NaN        NaN        NaN        NaN   

          11월        12월       합계(누계)  
4   24,725.00  24,852.00   304,152.00  
5   17,532.00  16,634.00   209,001.00  
6        0.71       0.67         0.69  
7   35,055.00  33,684.00   371,330.00  
8   18,325.00  17,515.00   223,110.00  
9        0.52       0.52         0.60  
10  34,898.00  30,181.00   

In [363]:
used_car_data.index

RangeIndex(start=4, stop=28, step=1)

In [364]:
used_car_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 24 entries, 4 to 27
Data columns (total 16 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   구분      24 non-null     object 
 1   연도      24 non-null     object 
 2   내역      24 non-null     object 
 3   1월      24 non-null     float64
 4   2월      24 non-null     float64
 5   3월      24 non-null     float64
 6   4월      24 non-null     float64
 7   5월      18 non-null     float64
 8   6월      18 non-null     float64
 9   7월      18 non-null     float64
 10  8월      18 non-null     float64
 11  9월      18 non-null     float64
 12  10월     18 non-null     float64
 13  11월     18 non-null     float64
 14  12월     18 non-null     float64
 15  합계(누계)  18 non-null     float64
dtypes: float64(13), object(3)
memory usage: 3.1+ KB


In [365]:
# 2024년 경매낙찰대수 경매출품대수

used_car_data.iloc[9, -1] = used_car_data.iloc[9, 3:].sum()

In [366]:
# 2024년 경매낙찰대수 낙찰대수

used_car_data.iloc[10, -1] = used_car_data.iloc[10, 3:].sum()

In [367]:
used_car_data

구분     연도        내역         1월         2월         3월         4월  \
4    경매낙찰대수  2021년    경매출품대수  25,419.00  22,800.00  24,334.00  28,117.00   
5    경매낙찰대수  2021년      낙찰대수  16,659.00  15,293.00  17,719.00  19,559.00   
6    경매낙찰대수  2021년    낙찰율(%)       0.66       0.67       0.73       0.70   
7    경매낙찰대수  2022년    경매출품대수  26,702.00  25,953.00  33,488.00  33,256.00   
8    경매낙찰대수  2022년      낙찰대수  17,852.00  17,551.00  21,481.00  19,243.00   
9    경매낙찰대수  2022년    낙찰율(%)       0.67       0.68       0.64       0.58   
10   경매낙찰대수  2023년    경매출품대수  31,238.00  29,794.00  33,702.00  29,912.00   
11   경매낙찰대수  2023년      낙찰대수  18,603.00  20,281.00  24,501.00  20,908.00   
12   경매낙찰대수  2023년  낙찰율 (당월)       0.60       0.68       0.73       0.70   
13   경매낙찰대수  2024년    경매출품대수  33,993.00  33,089.00  33,089.00  33,818.00   
14   경매낙찰대수  2024년      낙찰대수  20,284.00  21,609.00  21,609.00  21,217.00   
15   경매낙찰대수  2024년  낙찰율 (당월)       0.60       0.65       0.65       0.63   
16  중고차거래대수  2021년    사업자 거래 194,646.00 188,033.00 247,252.00 233,060.00   
17  중고차거래대수  2021년    당사자 거래 112,122.00 105,001.00 126,106.00 117,993.00   
18  중고차거래대수  2021년       총합계 306,768.00 293,034.00 373,358.00 351,053.00   
19  중고차거래대수  2022년    사업자 거래 200,042.00 195,899.00 226,655.00 217,975.00   
20  중고차거래대수  2022년    당사자 거래 100,239.00  94,242.00 107,611.00 103,954.00   
21  중고차거래대수  2022년       총합계 300,281.00 290,141.00 334,266.00 321,929.00   
22  중고차거래대수  2023년    사업자 거래 170,191.00 195,597.00 230,898.00 203,097.00   
23  중고차거래대수  2023년    당사자 거래 109,119.00 114,579.00 132,538.00 110,158.00   
24  중고차거래대수  2023년       총합계 279,310.00 310,176.00 363,436.00 313,255.00   
25  중고차거래대수  2024년    사업자 거래 217,457.00 190,954.00 223,854.00 228,623.00   
26  중고차거래대수  2024년    당사자 거래 118,339.00 102,041.00 113,047.00 114,387.00   
27  중고차거래대수  2024년       총합계 335,796.00 292,995.00 336,901.00 343,010.00   

           5월         6월         7월         8월         9월        10월  \
4   24,533.00  25,082.00  29,191.00  23,633.00  25,115.00  26,351.00   
5   17,032.00  17,843.00  19,860.00  16,209.00  16,676.00  17,985.00   
6        0.69       0.71       0.68       0.69       0.66       0.68   
7   30,134.00  31,563.00  33,369.00  30,573.00  29,416.00  28,137.00   
8   18,493.00  19,129.00  18,728.00  18,407.00  18,819.00  17,567.00   
9        0.61       0.61       0.56       0.60       0.64       0.62   
10  29,437.00  33,268.00  34,226.00  29,801.00  26,479.00  29,500.00   
11  20,011.00  22,187.00  20,944.00  20,166.00  17,901.00  19,030.00   
12       0.68       0.67       0.61       0.68       0.68       0.65   
13        NaN        NaN        NaN        NaN        NaN        NaN   
14        NaN        NaN        NaN        NaN        NaN        NaN   
15        NaN        NaN        NaN        NaN        NaN        NaN   
16 213,026.00 227,944.00 225,245.00 210,308.00 191,549.00 208,746.00   
17 104,710.00 113,129.00 110,018.00 101,569.00  93,120.00  99,908.00   
18 317,736.00 341,073.00 335,263.00 311,877.00 284,669.00 308,654.00   
19 224,118.00 210,354.00 209,396.00 217,584.00 202,374.00 200,684.00   
20 107,049.00 107,088.00 107,014.00 107,489.00 102,398.00 100,525.00   
21 331,167.00 317,442.00 316,410.00 325,073.00 304,772.00 301,209.00   
22 211,730.00 214,056.00 206,892.00 209,065.00 191,955.00 200,088.00   
23 111,048.00 115,704.00 108,253.00 108,751.00  96,711.00 102,817.00   
24 322,778.00 329,760.00 315,145.00 317,816.00 288,666.00 302,905.00   
25        NaN        NaN        NaN        NaN        NaN        NaN   
26        NaN        NaN        NaN        NaN        NaN        NaN   
27        NaN        NaN        NaN        NaN        NaN        NaN   

          11월        12월       합계(누계)  
4   24,725.00  24,852.00   304,152.00  
5   17,532.00  16,634.00   209,001.00  
6        0.71       0.67         0.69  
7   35,055.00  33,684.00   371,330.00  
8   18,325.00  17,515.00   223,110.00  
9        0.52       0.52         0.60  
10  34,898.00  30,181.00   

In [368]:
used_car_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 24 entries, 4 to 27
Data columns (total 16 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   구분      24 non-null     object 
 1   연도      24 non-null     object 
 2   내역      24 non-null     object 
 3   1월      24 non-null     float64
 4   2월      24 non-null     float64
 5   3월      24 non-null     float64
 6   4월      24 non-null     float64
 7   5월      18 non-null     float64
 8   6월      18 non-null     float64
 9   7월      18 non-null     float64
 10  8월      18 non-null     float64
 11  9월      18 non-null     float64
 12  10월     18 non-null     float64
 13  11월     18 non-null     float64
 14  12월     18 non-null     float64
 15  합계(누계)  20 non-null     float64
dtypes: float64(13), object(3)
memory usage: 3.1+ KB


In [369]:
setted_data=used_car_data.set_index(['연도'])

In [370]:
setted_data

구분        내역         1월         2월         3월         4월  \
연도                                                                     
2021년   경매낙찰대수    경매출품대수  25,419.00  22,800.00  24,334.00  28,117.00   
2021년   경매낙찰대수      낙찰대수  16,659.00  15,293.00  17,719.00  19,559.00   
2021년   경매낙찰대수    낙찰율(%)       0.66       0.67       0.73       0.70   
2022년   경매낙찰대수    경매출품대수  26,702.00  25,953.00  33,488.00  33,256.00   
2022년   경매낙찰대수      낙찰대수  17,852.00  17,551.00  21,481.00  19,243.00   
2022년   경매낙찰대수    낙찰율(%)       0.67       0.68       0.64       0.58   
2023년   경매낙찰대수    경매출품대수  31,238.00  29,794.00  33,702.00  29,912.00   
2023년   경매낙찰대수      낙찰대수  18,603.00  20,281.00  24,501.00  20,908.00   
2023년   경매낙찰대수  낙찰율 (당월)       0.60       0.68       0.73       0.70   
2024년   경매낙찰대수    경매출품대수  33,993.00  33,089.00  33,089.00  33,818.00   
2024년   경매낙찰대수      낙찰대수  20,284.00  21,609.00  21,609.00  21,217.00   
2024년   경매낙찰대수  낙찰율 (당월)       0.60       0.65       0.65       0.63   
2021년  중고차거래대수    사업자 거래 194,646.00 188,033.00 247,252.00 233,060.00   
2021년  중고차거래대수    당사자 거래 112,122.00 105,001.00 126,106.00 117,993.00   
2021년  중고차거래대수       총합계 306,768.00 293,034.00 373,358.00 351,053.00   
2022년  중고차거래대수    사업자 거래 200,042.00 195,899.00 226,655.00 217,975.00   
2022년  중고차거래대수    당사자 거래 100,239.00  94,242.00 107,611.00 103,954.00   
2022년  중고차거래대수       총합계 300,281.00 290,141.00 334,266.00 321,929.00   
2023년  중고차거래대수    사업자 거래 170,191.00 195,597.00 230,898.00 203,097.00   
2023년  중고차거래대수    당사자 거래 109,119.00 114,579.00 132,538.00 110,158.00   
2023년  중고차거래대수       총합계 279,310.00 310,176.00 363,436.00 313,255.00   
2024년  중고차거래대수    사업자 거래 217,457.00 190,954.00 223,854.00 228,623.00   
2024년  중고차거래대수    당사자 거래 118,339.00 102,041.00 113,047.00 114,387.00   
2024년  중고차거래대수       총합계 335,796.00 292,995.00 336,901.00 343,010.00   

              5월         6월         7월         8월         9월        10월  \
연도                                                                        
2021년  24,533.00  25,082.00  29,191.00  23,633.00  25,115.00  26,351.00   
2021년  17,032.00  17,843.00  19,860.00  16,209.00  16,676.00  17,985.00   
2021년       0.69       0.71       0.68       0.69       0.66       0.68   
2022년  30,134.00  31,563.00  33,369.00  30,573.00  29,416.00  28,137.00   
2022년  18,493.00  19,129.00  18,728.00  18,407.00  18,819.00  17,567.00   
2022년       0.61       0.61       0.56       0.60       0.64       0.62   
2023년  29,437.00  33,268.00  34,226.00  29,801.00  26,479.00  29,500.00   
2023년  20,011.00  22,187.00  20,944.00  20,166.00  17,901.00  19,030.00   
2023년       0.68       0.67       0.61       0.68       0.68       0.65   
2024년        NaN        NaN        NaN        NaN        NaN        NaN   
2024년        NaN        NaN        NaN        NaN        NaN        NaN   
2024년        NaN        NaN        NaN        NaN        NaN        NaN   
2021년 213,026.00 227,944.00 225,245.00 210,308.00 191,549.00 208,746.00   
2021년 104,710.00 113,129.00 110,018.00 101,569.00  93,120.00  99,908.00   
2021년 317,736.00 341,073.00 335,263.00 311,877.00 284,669.00 308,654.00   
2022년 224,118.00 210,354.00 209,396.00 217,584.00 202,374.00 200,684.00   
2022년 107,049.00 107,088.00 107,014.00 107,489.00 102,398.00 100,525.00   
2022년 331,167.00 317,442.00 316,410.00 325,073.00 304,772.00 301,209.00   
2023년 211,730.00 214,056.00 206,892.00 209,065.00 191,955.00 200,088.00   
2023년 111,048.00 115,704.00 108,253.00 108,751.00  96,711.00 102,817.00   
2023년 322,778.00 329,760.00 315,145.00 317,816.00 288,666.00 302,905.00   
2024년        NaN        NaN        NaN        NaN        NaN        NaN   
2024년        NaN        NaN        NaN        NaN        NaN        NaN   
2024년        NaN        NaN        NaN        NaN        NaN        NaN   

             11월        12월       합계(누계)  
연도                                        
2021년  24,725.00  24,852.00   304,152.00  
2021년  17,532.00  16,634.00   209,001.00  
2021년       0.71

In [371]:
doubled_setted_data = used_car_data.set_index(['연도', '내역'])

In [372]:
pd.options.display.float_format = '{:,.2f}'.format

In [373]:
doubled_setted_data

구분         1월         2월         3월         4월  \
연도    내역                                                              
2021년 경매출품대수     경매낙찰대수  25,419.00  22,800.00  24,334.00  28,117.00   
      낙찰대수       경매낙찰대수  16,659.00  15,293.00  17,719.00  19,559.00   
      낙찰율(%)     경매낙찰대수       0.66       0.67       0.73       0.70   
2022년 경매출품대수     경매낙찰대수  26,702.00  25,953.00  33,488.00  33,256.00   
      낙찰대수       경매낙찰대수  17,852.00  17,551.00  21,481.00  19,243.00   
      낙찰율(%)     경매낙찰대수       0.67       0.68       0.64       0.58   
2023년 경매출품대수     경매낙찰대수  31,238.00  29,794.00  33,702.00  29,912.00   
      낙찰대수       경매낙찰대수  18,603.00  20,281.00  24,501.00  20,908.00   
      낙찰율 (당월)   경매낙찰대수       0.60       0.68       0.73       0.70   
2024년 경매출품대수     경매낙찰대수  33,993.00  33,089.00  33,089.00  33,818.00   
      낙찰대수       경매낙찰대수  20,284.00  21,609.00  21,609.00  21,217.00   
      낙찰율 (당월)   경매낙찰대수       0.60       0.65       0.65       0.63   
2021년 사업자 거래    중고차거래대수 194,646.00 188,033.00 247,252.00 233,060.00   
      당사자 거래    중고차거래대수 112,122.00 105,001.00 126,106.00 117,993.00   
      총합계       중고차거래대수 306,768.00 293,034.00 373,358.00 351,053.00   
2022년 사업자 거래    중고차거래대수 200,042.00 195,899.00 226,655.00 217,975.00   
      당사자 거래    중고차거래대수 100,239.00  94,242.00 107,611.00 103,954.00   
      총합계       중고차거래대수 300,281.00 290,141.00 334,266.00 321,929.00   
2023년 사업자 거래    중고차거래대수 170,191.00 195,597.00 230,898.00 203,097.00   
      당사자 거래    중고차거래대수 109,119.00 114,579.00 132,538.00 110,158.00   
      총합계       중고차거래대수 279,310.00 310,176.00 363,436.00 313,255.00   
2024년 사업자 거래    중고차거래대수 217,457.00 190,954.00 223,854.00 228,623.00   
      당사자 거래    중고차거래대수 118,339.00 102,041.00 113,047.00 114,387.00   
      총합계       중고차거래대수 335,796.00 292,995.00 336,901.00 343,010.00   

                       5월         6월         7월         8월         9월  \
연도    내역                                                                
2021년 경매출품대수    24,533.00  25,082.00  29,191.00  23,633.00  25,115.00   
      낙찰대수      17,032.00  17,843.00  19,860.00  16,209.00  16,676.00   
      낙찰율(%)         0.69       0.71       0.68       0.69       0.66   
2022년 경매출품대수    30,134.00  31,563.00  33,369.00  30,573.00  29,416.00   
      낙찰대수      18,493.00  19,129.00  18,728.00  18,407.00  18,819.00   
      낙찰율(%)         0.61       0.61       0.56       0.60       0.64   
2023년 경매출품대수    29,437.00  33,268.00  34,226.00  29,801.00  26,479.00   
      낙찰대수      20,011.00  22,187.00  20,944.00  20,166.00  17,901.00   
      낙찰율 (당월)       0.68       0.67       0.61       0.68       0.68   
2024년 경매출품대수          NaN        NaN        NaN        NaN        NaN   
      낙찰대수            NaN        NaN        NaN        NaN        NaN   
      낙찰율 (당월)        NaN        NaN        NaN        NaN        NaN   
2021년 사업자 거래   213,026.00 227,944.00 225,245.00 210,308.00 191,549.00   
      당사자 거래   104,710.00 113,129.00 110,018.00 101,569.00  93,120.00   
      총합계      317,736.00 341,073.00 335,263.00 311,877.00 284,669.00   
2022년 사업자 거래   224,118.00 210,354.00 209,396.00 217,584.00 202,374.00   
      당사자 거래   107,049.00 107,088.00 107,014.00 107,489.00 102,398.00   
      총합계      331,167.00 317,442.00 316,410.00 325,073.00 304,772.00   
2023년 사업자 거래   211,730.00 214,056.00 206,892.00 209,065.00 191,955.00   
      당사자 거래   111,048.00 115,704.00 108,253.00 108,751.00  96,711.00   
      총합계      322,778.00 329,760.00 315,145.00 317,816.00 288,666.00   
2024년 사업자 거래          NaN        NaN        NaN        NaN        NaN   
      당사자 거래          NaN        NaN        NaN        NaN        NaN   
      총합계             NaN        NaN        NaN        NaN        NaN   

                      10월        11월        12월       합계(누계)  
연도    내역                                                      
2021년 경매출품대수    26,351.00  24,725.00  24,852.00   304,152.00  
      낙찰대수      17,985.00  17,532.00  16,634.00   209,001.00  
      낙찰율(%)         0.

In [374]:
doubled_setted_data.info()

<class 'pandas.core.frame.DataFrame'>
MultiIndex: 24 entries, ('2021년', '경매출품대수') to ('2024년', '총합계')
Data columns (total 14 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   구분      24 non-null     object 
 1   1월      24 non-null     float64
 2   2월      24 non-null     float64
 3   3월      24 non-null     float64
 4   4월      24 non-null     float64
 5   5월      18 non-null     float64
 6   6월      18 non-null     float64
 7   7월      18 non-null     float64
 8   8월      18 non-null     float64
 9   9월      18 non-null     float64
 10  10월     18 non-null     float64
 11  11월     18 non-null     float64
 12  12월     18 non-null     float64
 13  합계(누계)  20 non-null     float64
dtypes: float64(13), object(1)
memory usage: 3.4+ KB


In [375]:
doubled_setted_data.loc[('2021년','경매출품대수')]

구분           경매낙찰대수
1월        25,419.00
2월        22,800.00
3월        24,334.00
4월        28,117.00
5월        24,533.00
6월        25,082.00
7월        29,191.00
8월        23,633.00
9월        25,115.00
10월       26,351.00
11월       24,725.00
12월       24,852.00
합계(누계)   304,152.00
Name: (2021년, 경매출품대수), dtype: object

In [376]:
doubled_setted_data.loc[('2022년','경매출품대수')]

구분           경매낙찰대수
1월        26,702.00
2월        25,953.00
3월        33,488.00
4월        33,256.00
5월        30,134.00
6월        31,563.00
7월        33,369.00
8월        30,573.00
9월        29,416.00
10월       28,137.00
11월       35,055.00
12월       33,684.00
합계(누계)   371,330.00
Name: (2022년, 경매출품대수), dtype: object

In [377]:
doubled_setted_data.loc[('2023년','경매출품대수')]

구분           경매낙찰대수
1월        31,238.00
2월        29,794.00
3월        33,702.00
4월        29,912.00
5월        29,437.00
6월        33,268.00
7월        34,226.00
8월        29,801.00
9월        26,479.00
10월       29,500.00
11월       34,898.00
12월       30,181.00
합계(누계)   372,436.00
Name: (2023년, 경매출품대수), dtype: object

In [378]:
print('2021년, 경매출품대수 총합', doubled_setted_data.loc[('2021년','경매출품대수')][-1])
print('2022년, 경매출품대수 총합', doubled_setted_data.loc[('2022년','경매출품대수')][-1])
print('2023년, 경매출품대수 총합', doubled_setted_data.loc[('2023년','경매출품대수')][-1])

2021년, 경매출품대수 총합 304152.0
2022년, 경매출품대수 총합 371330.0
2023년, 경매출품대수 총합 372436.0


C:\Users\KDP-50\AppData\Local\Temp\ipykernel_11004\4289266403.py:1: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  print('2021년, 경매출품대수 총합', doubled_setted_data.loc[('2021년','경매출품대수')][-1])
C:\Users\KDP-50\AppData\Local\Temp\ipykernel_11004\4289266403.py:2: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  print('2022년, 경매출품대수 총합', doubled_setted_data.loc[('2022년','경매출품대수')][-1])
C:\Users\KDP-50\AppData\Local\Temp\ipykernel_11004\4289266403.py:3: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior

In [379]:
doubled_setted_data.loc[('2021년','낙찰대수')]

구분           경매낙찰대수
1월        16,659.00
2월        15,293.00
3월        17,719.00
4월        19,559.00
5월        17,032.00
6월        17,843.00
7월        19,860.00
8월        16,209.00
9월        16,676.00
10월       17,985.00
11월       17,532.00
12월       16,634.00
합계(누계)   209,001.00
Name: (2021년, 낙찰대수), dtype: object

In [380]:
doubled_setted_data.loc[('2022년','낙찰대수')]

구분           경매낙찰대수
1월        17,852.00
2월        17,551.00
3월        21,481.00
4월        19,243.00
5월        18,493.00
6월        19,129.00
7월        18,728.00
8월        18,407.00
9월        18,819.00
10월       17,567.00
11월       18,325.00
12월       17,515.00
합계(누계)   223,110.00
Name: (2022년, 낙찰대수), dtype: object

In [381]:
doubled_setted_data.loc[('2023년','낙찰대수')]

구분           경매낙찰대수
1월        18,603.00
2월        20,281.00
3월        24,501.00
4월        20,908.00
5월        20,011.00
6월        22,187.00
7월        20,944.00
8월        20,166.00
9월        17,901.00
10월       19,030.00
11월       21,237.00
12월       18,314.00
합계(누계)   244,083.00
Name: (2023년, 낙찰대수), dtype: object

In [382]:
print('2021년, 낙찰대수 총합', doubled_setted_data.loc[('2021년','낙찰대수')][-1])
print('2022년, 낙찰대수 총합', doubled_setted_data.loc[('2022년','낙찰대수')][-1])
print('2023년, 낙찰대수 총합', doubled_setted_data.loc[('2023년','낙찰대수')][-1])

2021년, 낙찰대수 총합 209001.0
2022년, 낙찰대수 총합 223110.0
2023년, 낙찰대수 총합 244083.0


C:\Users\KDP-50\AppData\Local\Temp\ipykernel_11004\1959113155.py:1: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  print('2021년, 낙찰대수 총합', doubled_setted_data.loc[('2021년','낙찰대수')][-1])
C:\Users\KDP-50\AppData\Local\Temp\ipykernel_11004\1959113155.py:2: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  print('2022년, 낙찰대수 총합', doubled_setted_data.loc[('2022년','낙찰대수')][-1])
C:\Users\KDP-50\AppData\Local\Temp\ipykernel_11004\1959113155.py:3: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To ac

In [383]:
doubled_setted_data.loc[('2021년','낙찰율(%)')]

구분        경매낙찰대수
1월          0.66
2월          0.67
3월          0.73
4월          0.70
5월          0.69
6월          0.71
7월          0.68
8월          0.69
9월          0.66
10월         0.68
11월         0.71
12월         0.67
합계(누계)      0.69
Name: (2021년, 낙찰율(%)), dtype: object

In [384]:
doubled_setted_data.loc[('2022년','낙찰율(%)')]

구분        경매낙찰대수
1월          0.67
2월          0.68
3월          0.64
4월          0.58
5월          0.61
6월          0.61
7월          0.56
8월          0.60
9월          0.64
10월         0.62
11월         0.52
12월         0.52
합계(누계)      0.60
Name: (2022년, 낙찰율(%)), dtype: object

In [385]:
doubled_setted_data.loc[('2023년','낙찰율 (당월)')]

구분        경매낙찰대수
1월          0.60
2월          0.68
3월          0.73
4월          0.70
5월          0.68
6월          0.67
7월          0.61
8월          0.68
9월          0.68
10월         0.65
11월         0.61
12월         0.61
합계(누계)      0.66
Name: (2023년, 낙찰율 (당월)), dtype: object

In [386]:
print('2021년 - 낙찰율(%)', doubled_setted_data.loc[('2021년','낙찰율(%)')][-1])
print('2022년 - 낙찰율(%)', doubled_setted_data.loc[('2022년','낙찰율(%)')][-1])
print('2023년 - 낙찰율(%)', doubled_setted_data.loc[('2023년','낙찰율 (당월)')][-1])

2021년 - 낙찰율(%) 0.6871597096188747
2022년 - 낙찰율(%) 0.600840222982253
2023년 - 낙찰율(%) 0.6553689761462372


C:\Users\KDP-50\AppData\Local\Temp\ipykernel_11004\1167467821.py:1: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  print('2021년 - 낙찰율(%)', doubled_setted_data.loc[('2021년','낙찰율(%)')][-1])
C:\Users\KDP-50\AppData\Local\Temp\ipykernel_11004\1167467821.py:2: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  print('2022년 - 낙찰율(%)', doubled_setted_data.loc[('2022년','낙찰율(%)')][-1])
C:\Users\KDP-50\AppData\Local\Temp\ipykernel_11004\1167467821.py:3: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). T

In [387]:
doubled_setted_data.loc[('2021년','사업자 거래')]

구분            중고차거래대수
1월         194,646.00
2월         188,033.00
3월         247,252.00
4월         233,060.00
5월         213,026.00
6월         227,944.00
7월         225,245.00
8월         210,308.00
9월         191,549.00
10월        208,746.00
11월        217,818.00
12월        214,706.00
합계(누계)   2,572,333.00
Name: (2021년, 사업자 거래), dtype: object

In [388]:
doubled_setted_data.loc[('2022년','사업자 거래')]

구분            중고차거래대수
1월         200,042.00
2월         195,899.00
3월         226,655.00
4월         217,975.00
5월         224,118.00
6월         210,354.00
7월         209,396.00
8월         217,584.00
9월         202,374.00
10월        200,684.00
11월        194,530.00
12월        168,909.00
합계(누계)   2,468,520.00
Name: (2022년, 사업자 거래), dtype: object

In [389]:
doubled_setted_data.loc[('2023년','사업자 거래')]

구분            중고차거래대수
1월         170,191.00
2월         195,597.00
3월         230,898.00
4월         203,097.00
5월         211,730.00
6월         214,056.00
7월         206,892.00
8월         209,065.00
9월         191,955.00
10월        200,088.00
11월        214,114.00
12월        192,221.00
합계(누계)   4,983,778.00
Name: (2023년, 사업자 거래), dtype: object

In [390]:
print('2021년 - 사업자 거래', doubled_setted_data.loc[('2021년','사업자 거래')][-1])
print('2022년 - 사업자 거래', doubled_setted_data.loc[('2022년','사업자 거래')][-1])
print('2023년 - 사업자 거래', doubled_setted_data.loc[('2023년','사업자 거래')][-1])

2021년 - 사업자 거래 2572333.0
2022년 - 사업자 거래 2468520.0
2023년 - 사업자 거래 4983778.0


C:\Users\KDP-50\AppData\Local\Temp\ipykernel_11004\2555445063.py:1: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  print('2021년 - 사업자 거래', doubled_setted_data.loc[('2021년','사업자 거래')][-1])
C:\Users\KDP-50\AppData\Local\Temp\ipykernel_11004\2555445063.py:2: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  print('2022년 - 사업자 거래', doubled_setted_data.loc[('2022년','사업자 거래')][-1])
C:\Users\KDP-50\AppData\Local\Temp\ipykernel_11004\2555445063.py:3: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). T

In [391]:
# 이 파트에서 2023년 사업자 거래가 4,983,778대라고 확인이 되었으나 이상하게 생각되어서
# Raw 데이터를 다시 봤더니 계산이 잘못되었음을 뒤늦게 확인하게 되었음.

# 데이터를 사용하기 전에 먼저 올바른 데이터인지 확인하는 
# 데이터 검증 단계가 필요할 것이라 생각하게됨.


sum_range=doubled_setted_data.loc[('2023년','사업자 거래')][:-1][1:]
total=0
for i in sum_range:
    i=int(i)
    total+=i
total

2439904

In [392]:
print('2021년 - 사업자 거래', doubled_setted_data.loc[('2021년','사업자 거래')][-1])
print('2022년 - 사업자 거래', doubled_setted_data.loc[('2022년','사업자 거래')][-1])
print('2023년 - (실) 사업자 거래', total)

2021년 - 사업자 거래 2572333.0
2022년 - 사업자 거래 2468520.0
2023년 - (실) 사업자 거래 2439904


C:\Users\KDP-50\AppData\Local\Temp\ipykernel_11004\713717689.py:1: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  print('2021년 - 사업자 거래', doubled_setted_data.loc[('2021년','사업자 거래')][-1])
C:\Users\KDP-50\AppData\Local\Temp\ipykernel_11004\713717689.py:2: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  print('2022년 - 사업자 거래', doubled_setted_data.loc[('2022년','사업자 거래')][-1])


In [393]:
doubled_setted_data.loc[('2021년','당사자 거래')]

구분            중고차거래대수
1월         112,122.00
2월         105,001.00
3월         126,106.00
4월         117,993.00
5월         104,710.00
6월         113,129.00
7월         110,018.00
8월         101,569.00
9월          93,120.00
10월         99,908.00
11월        104,061.00
12월        112,251.00
합계(누계)   1,299,988.00
Name: (2021년, 당사자 거래), dtype: object

In [394]:
doubled_setted_data.loc[('2022년','사업자 거래')]

구분            중고차거래대수
1월         200,042.00
2월         195,899.00
3월         226,655.00
4월         217,975.00
5월         224,118.00
6월         210,354.00
7월         209,396.00
8월         217,584.00
9월         202,374.00
10월        200,684.00
11월        194,530.00
12월        168,909.00
합계(누계)   2,468,520.00
Name: (2022년, 사업자 거래), dtype: object

In [395]:
doubled_setted_data.loc[('2023년','당사자 거래')]

구분            중고차거래대수
1월         109,119.00
2월         114,579.00
3월         132,538.00
4월         110,158.00
5월         111,048.00
6월         115,704.00
7월         108,253.00
8월         108,751.00
9월          96,711.00
10월        102,817.00
11월        108,858.00
12월        105,227.00
합계(누계)   8,709,927.00
Name: (2023년, 당사자 거래), dtype: object

In [396]:
print('2021년 - 당사자 거래', doubled_setted_data.loc[('2021년','당사자 거래')][-1])
print('2022년 - 당사자 거래', doubled_setted_data.loc[('2022년','당사자 거래')][-1])
print('2023년 - 당사자 거래', doubled_setted_data.loc[('2023년','당사자 거래')][-1])

2021년 - 당사자 거래 1299988.0
2022년 - 당사자 거래 1257629.0
2023년 - 당사자 거래 8709927.0


C:\Users\KDP-50\AppData\Local\Temp\ipykernel_11004\1864422298.py:1: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  print('2021년 - 당사자 거래', doubled_setted_data.loc[('2021년','당사자 거래')][-1])
C:\Users\KDP-50\AppData\Local\Temp\ipykernel_11004\1864422298.py:2: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  print('2022년 - 당사자 거래', doubled_setted_data.loc[('2022년','당사자 거래')][-1])
C:\Users\KDP-50\AppData\Local\Temp\ipykernel_11004\1864422298.py:3: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). T

In [397]:
# 마찬가지로 Raw데이터의 2023년 당사자 거래량 총량도 8,709,927대로 이상하여
# Raw 데이터를 다시 봤더니 계산이 잘못되었음.

sum_range2=doubled_setted_data.loc[('2023년','당사자 거래')][:-1][1:]
total2=0
for i in sum_range2:
    i=int(i)
    total2+=i
total2

1323763

In [398]:
print('2021년 - 당사자 거래', doubled_setted_data.loc[('2021년','당사자 거래')][-1])
print('2022년 - 당사자 거래', doubled_setted_data.loc[('2022년','당사자 거래')][-1])
print('2023년 - 당사자 거래', total2)

2021년 - 당사자 거래 1299988.0
2022년 - 당사자 거래 1257629.0
2023년 - 당사자 거래 1323763


C:\Users\KDP-50\AppData\Local\Temp\ipykernel_11004\567211560.py:1: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  print('2021년 - 당사자 거래', doubled_setted_data.loc[('2021년','당사자 거래')][-1])
C:\Users\KDP-50\AppData\Local\Temp\ipykernel_11004\567211560.py:2: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  print('2022년 - 당사자 거래', doubled_setted_data.loc[('2022년','당사자 거래')][-1])


In [399]:
# cpi와의 대조작업을 진행.

check_cpi_3years = check_cpi.iloc[0]

In [400]:
check_cpi_3years

2021   102.50
2022   107.70
2023   111.60
Name: 소비자물가 총지수(2020=100), dtype: float64

In [401]:
year_names_columns=['2021년','2022년','2023년']
check_cpi_3years.index = year_names_columns

In [402]:
check_cpi_3years

2021년   102.50
2022년   107.70
2023년   111.60
Name: 소비자물가 총지수(2020=100), dtype: float64

In [403]:
# 경매시장 데이터 가져오기

used_total21=doubled_setted_data.loc[('2021년','경매출품대수')][-1]
used_total22=doubled_setted_data.loc[('2022년','경매출품대수')][-1]
used_total23=doubled_setted_data.loc[('2023년','경매출품대수')][-1]

buy_used21=doubled_setted_data.loc[('2021년','낙찰대수')][-1]
buy_used22=doubled_setted_data.loc[('2022년','낙찰대수')][-1]
buy_used23=doubled_setted_data.loc[('2023년','낙찰대수')][-1]

company_buy21=doubled_setted_data.loc[('2021년','사업자 거래')][-1]
company_buy22=doubled_setted_data.loc[('2022년','사업자 거래')][-1]
company_buy23=doubled_setted_data.loc[('2023년','사업자 거래')][-1]

person_buy21=doubled_setted_data.loc[('2021년','당사자 거래')][-1]
person_buy22=doubled_setted_data.loc[('2022년','당사자 거래')][-1]
person_buy23=doubled_setted_data.loc[('2023년','당사자 거래')][-1]

C:\Users\KDP-50\AppData\Local\Temp\ipykernel_11004\1932891048.py:3: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  used_total21=doubled_setted_data.loc[('2021년','경매출품대수')][-1]
C:\Users\KDP-50\AppData\Local\Temp\ipykernel_11004\1932891048.py:4: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  used_total22=doubled_setted_data.loc[('2022년','경매출품대수')][-1]
C:\Users\KDP-50\AppData\Local\Temp\ipykernel_11004\1932891048.py:5: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by posi

In [404]:
usedcarDF=pd.DataFrame({'경매출품대수':[used_total21, used_total22, used_total23],
              '낙찰대수':[buy_used21, buy_used22, buy_used23],
              '사업자 거래':[company_buy21, company_buy22, total],
              '당사자 거래':[person_buy21, person_buy22, total2]}, index=year_names_columns)
usedcarDF

경매출품대수       낙찰대수       사업자 거래       당사자 거래
2021년 304,152.00 209,001.00 2,572,333.00 1,299,988.00
2022년 371,330.00 223,110.00 2,468,520.00 1,257,629.00
2023년 372,436.00 244,083.00 2,439,904.00 1,323,763.00

In [405]:
usedcarDF.join(check_cpi_3years)

경매출품대수       낙찰대수       사업자 거래       당사자 거래  소비자물가 총지수(2020=100)
2021년 304,152.00 209,001.00 2,572,333.00 1,299,988.00               102.50
2022년 371,330.00 223,110.00 2,468,520.00 1,257,629.00               107.70
2023년 372,436.00 244,083.00 2,439,904.00 1,323,763.00               111.60

#### (결론)

- 해마다 소비자물가 총지수가 오름에 따라  
  중고차 경매시장 데이터를 보면  

- 경매출품대수는  
  2021년  
  304,152대 

  2022년  
  371,330대  

  2023년  
  372,436대 이므로  
  해마다 중고차 경매출품대수가 늘었음을 알 수 있었고,  

- 경매낙찰대수 또한  
  2021년  
  209,001대  
  2022년  
  223,110대  
  2023년  
  244,083대 이므로  
  해마다 중고차 경매낙찰대수 또한 늘었음을 알 수 있었음.  

- 사업자 거래와 당사자 거래를 토대로 비교해보았을 때는 

  2021년  
  2,572,333대  
  2022년  
  2,468,520대  
  2023년  
  2,439,904대  

  2021년  
  1,299,988대  
  2022년  
  1,257,629대  
  2023년  
  1,323,763대  

- 앞서 증가하던 경매출품대수와 경매낙찰대수와 대조적으로  
  사업자 거래량은 오히려 감소했고,  
  당사자 거래량은 증감이 불규칙함.

- 해마다 중고차 경매출품대수와 경매낙찰대수가 느는 것은 소비자물가 총지수가 오름에 따라  
  소비자 부담으로 작용하여 중고차 경매에 대한 선호도가 높아지는 것으로 사료됨.